Release Godot 3.6 beta 4 and publish release notes (#788)
godotengine/godot-website: collections/_article/dev-snapshot-godot-3-6-beta-4.md

---
title: "Dev snapshot: Godot 3.6 beta 4"

> excerpt: ""

Reviewer: Missing excerpt :)

categories: ["pre-release"]
author: Rémi Verschelde
image: /storage/blog/covers/dev-snapshot-godot-3-6-beta-4.webp
image_caption_title: "Cardbob"
image_caption_description: "A game by Mad Cookies Studio"
date: 2024-01-25 13:00:00
---

Long time no see, Godot 3.x! While our main focus is on Godot 4.x with frequent feature releases (with the recent release of [Godot 4.2](https://godotengine.org/article/godot-4-2-arrives-in-style/) in November), we still have a number of changes backported to the `3.x` branch or written specifically for that version, which we plan to release as Godot 3.6 in the near future.

Godot 3 is still very important for a significant part of the ecosystem – notably people with released games, games close to being released, or who need to target OpenGL 2 / WebGL 1 to maximize device compatibility.

> The last 3.6 beta snapshot was [back in August](/article/dev-snapshot-godot-3-6-beta-3/), so this new one is somewhat overdue... but better late than never, and it includes a lot of important bug fixes to platform compatibility, rendering, or thirdparty libraries. This beta 4 should be a good baseline to see where we stand in the 3.6 development, and plan what's left before the complete feature freeze and release candidate stage.

Reviewer: ```suggestion
The last 3.6 beta snapshot was [back in August](/article/dev-snapshot-godot-3-6-beta-3/), so this new one is somewhat overdue... but better late than never, and it includes a lot of important bug fixes to platform compatibility, rendering, and thirdparty libraries. This beta 4 should be a good baseline to see where we stand in the 3.6 development, and plan what's left before the complete feature freeze and release candidate stage.
```

"or" feels weird here.

### Updated toolchains for official builds

We've updated the toolchains (compiler, platform SDKs, etc.) for this release to match the versions used for Godot 4.2 ([build-containers#135](https://github.com/godotengine/build-containers/pull/135)).
This notably gives us the following toolchains:

- Base image: Fedora 39
- Mono version: 6.12.0.198
- SCons: 4.5.2
- Linux: GCC 13.2.0 built against glibc 2.28, binutils 2.40, from our own Linux SDK
- Windows: MinGW 11.0.0, GCC 13.2.1, binutils 2.40
- HTML5: Emscripten 3.1.39 (standard builds), Emscripten 1.39.9 (Mono builds)
- Android: Android NDK 23.2.8568313, build-tools 33.0.2, platform android-33, CMake 3.22.1, JDK 11
- macOS: Xcode 15.0 with Apple Clang (LLVM 16.0.0), MacOSX SDK 14.0
- iOS: Xcode 15.0 with Apple Clang (LLVM 16.0.0), iPhoneOS SDK 17.0
- UWP: Visual Studio 2017 (unchanged)

In practice this shouldn't change much, aside from getting some better compile and linking-time optimizations thanks to newer toolchain versions, leading to potential size or performance improvements. And some bug fixes which may or may not affect our builds.

### Official Linux ARM builds

> [Like we added in Godot 4.2](https://godotengine.org/article/godot-4-2-arrives-in-style/#linux), we now have official Linux ARM builds (`arm32` and `arm64`). This should allow running both the Godot editor on ARM devices such as Raspberry Pi or the Linux subsystem on Chromebooks, and exporting Godot projects to them. The Linux export template now lets you select the architecture at export time among the four options supported in 3.6: `x86_64` (default), `x86_32`, `arm64`, `arm32`.

Reviewer: ```suggestion
[Just like Godot 4.2](https://godotengine.org/article/godot-4-2-arrives-in-style/#linux), we now have official Linux ARM builds (`arm32` and `arm64`) of Godot 3.6. This should allow both running the Godot editor on ARM devices (such as Raspberry Pi) and Chromebooks with the Linux subsystem, and exporting Godot projects to them. The Linux export template now lets you select the architecture at export time among the four options supported in 3.6: `x86_64` (default), `x86_32`, `arm64`, `arm32`.
```

### Update to minimal supported versions for Apple platforms

To ensure compatibility with current Apple SDKs, we had to increase the minimal supported versions from iOS 10 to iOS 12 ([GH-87359](https://github.com/godotengine/godot/pull/87359)), and from macOS 10.12 to macOS 10.13 for `x86_64` ([GH-76394](https://github.com/godotengine/godot/pull/76394)).

### Lightprobes and directional shadows performance improvements

> Among a number of other bugfixes, a couple performance improvements have been added for BakedLightmap lightprobes ([GH-80764](https://github.com/godotengine/godot/pull/80764)) and directional shadows ([GH-75468](https://github.com/godotengine/godot/pull/75468)).

Reviewer: Should "lightprobes" be spelled as two words to make it less jargony?
